grails-app/domain/org/pih/warehouse/order/OrderAdjustment.groovy

```diff
@@ -108,6 +110,14 @@ class OrderAdjustment implements Serializable, Comparable<OrderAdjustment> {
         return invoiceItems*.invoice.unique()
     }
 
+    Boolean getIsInvoiceable() {
+        return !isInvoiced && order.isPlaced()
+    }
+
+    Boolean getIsEncumbered() {
+        return !isInvoiced && hasPrepaymentInvoice
+    }
+
```

Comment:
I know we've done that in some places, but I don't like the naming like `getIs`.
Booleans should be named as `isSomething`.
Moreover, those don't have to be marked transients anymore, so you could just have a method `isInvoicable()`.

Reply:
Yeah, but the naming convention with get, allows us to use the `.getIsEncumbered()` in this way: `.isEncumbered`, so I prefer this groovy syntactic sugar

Reply:
I don't see anything bad in using `.isEncumbered()` vs forcing to access a property via the field name (`isEncumbered`). I feel like naming booleans using `getIs`, `getHas` is an anti-pattern we should avoid.
https://rules.sonarsource.com/java/tag/convention/RSPEC-2047/

Reply:
This is groovy, not Java :smile: 
The rule that you send is another case. They say that we should use `hasBar`, not `getBar`

Reply:
yeah, and why should we use `getBar` for a boolean field?

Reply:
not `getBar`, but `getIsBar`, and then we can use it `isBar`

Reply:
I will repeat it again, and you will do whatever you want - `getIs` feels like antipattern, and I don't see anything wrong in using `isBar()` in this case.

Reply:
I'm used to seeing fields like:
```
invoiceable
encumbered
```
with getters and setters like
```
isInvoiceable
setInvoiceable
isEncumbered
setEncumbered
```
From the groovy docs: https://groovy-lang.org/objectorientation.html#_property_naming_conventions

> ...generated getters and setters will have a name formed by capitalizing the property name and adding a get or set prefix (or optionally "is" for a boolean getter)... isEmpty might be the getter method name for a property named empty.

So with that setup in Groovy we could do `boolean isOrderInvoiceable = order.isInvoiceable()` as Kacper says. That's my personal preference but if you disagree Alan feel free to do as you wish.

Reply:
Java beans use the same convention. 

So since we keep having this conversation 

https://github.com/openboxes/openboxes/pull/4266#pullrequestreview-1627749462

... let's just use the java/groovy convention from now one. 

So no more "is" prefix in the property name. 

For me, I prefer using Groovy's implicit getter method
```
if (order.invoiceable)
```
over the explicit getter methods
```
if(order.getInvoiceable())
```
or 
```
if(order.isInvoiceable())
```

Reply:
The problem with `getIs` is that the property itself would be named with the `is` prefix, e.g. `isInvoiceable` instead of `invoiceable`. 
https://stackoverflow.com/a/5322730

> Your field should not be called isCurrent in the first place. Is is a verb and verbs are inappropriate to represent an Object's state. Use an adjective instead, and suddenly your getter / setter names will make more sense: